egs2/TEMPLATE/enh1/README.md

```diff
@@ -0,0 +1,90 @@
+# Speech Enhancement Frontend Recipe
+
+This is the common recipe for for ESPnet2 speech enhancement frontend. Currently, two speech separation recipes (`wsj0_2mix` and `wsj0_2mix_spatialized`) are supported:
+```
+egs2/
+├── wsj0_2mix
+│   └── enh1
+└── wsj0_2mix_spatialized
+    └── enh1
+```
+In `egs2/TEMPLATE/enh1/enh.sh`, 9 stages are included.
+
+#### Stage 1: Data preparation
+This stage is same as stage 1 in asr.sh.
+
+#### Stage 2: Speech perturbation
+Speech perturbation is widely used in ASR task, but rarely in speech enhancement. Some of our initial experiments have shown that speech perturbation works on `wsj0_2mix`. We are conducting more experiment to make sure if it works.
+The speech perturbation procedure is almost same as that in ASR, I have copied `scripts/utils/perturb_data_dir_speed.sh` to `scripts/utils/perturb_enh_data_dir_speed.sh` and made minor modification to support the speech perturbation for more scp files rather than `wav.scp` only.
+
+#### Stage 3: Format wav.scp
+Format scp files such as `wav.scp`. The scp files include:
+  + `wav.scp`: wav file list of mixed/noisy input signals.
+  + `spk{}.scp`: wav file list of speech reference signals. {} can be 1, 2, ..., depending on the number of speakers in the input signal in `wav.scp`.
+  + `noise{}.scp` (optional): wav file list of noise reference signals. {} can be 1, 2, ..., depending on the number of noise types in the input signal in `wav.scp`. The file(s) are required when `--use_noise_ref true` is specified.
+  + `dereverb.scp` (optional): wav file list of dereverberation reference signals (for training a dereverberation model). This file is required when `--use_dereverb_ref true` is specified.
+
+#### Stage 4: Remove short data
+This stage is same as that in ASR recipe.
+
+#### Stage 5: Collect stats for enhancement task.
+Same as the ASR task, we collet the data stats before training. Related new python files are:
+```
+espnet2/
+├── bin
+│   └── enh_train.py
+└── tasks
+    └── frontend.py
+```
+The `FrontendTask` defined in `frontend.py` is called in `enh_train.py`. In the `collect_stats` mode. the behavior of `FrontendTask` is same as the `ABSTask`.
+
+#### Stage 6: Enhancemnt Frontend Training
+We have created `FrontendTask` in `espnet2/tasks/frontend.py`, which is used to train `EnhFrontend(AbsFrontend)` that is defined in `espnet2/asr/frontend/enh_frontend.py` . Although it is currently defined as an independent task, the `EnhFrontend` is inherited from `AbsFrontend` and can be easily called by ASR tasks or even jointly trained with ASR in the future. The `ESPnetFrontendModel` defined in `espnet2/asr/frontend/espnet_model.py` is a wrapper of `EnhFrontend`.
```

Comment:
This part must be modified according to our recent changes

Reply:
Done.

Reply:
Done.